train/tcn_train.py

```diff
@@ -0,0 +1,564 @@
+import glob
+import math
+
+from torch import nn, optim
+from torch.utils.data import DataLoader
+from datetime import timedelta, datetime
+
+from datasets.labels import classes, labels
+from datasets.inception_subset import Modalities, InceptionSubset
+from datasets.experiments import experiments_info
+from datasets.transformations import *
+from model.PlantFeatExtNew import PlantFeatureExtractor
+
+from train.utils import get_checkpoint_name, get_training_name
+from train import parameters  # importing all parameters/home/pnn/PNN/train
+
+import matplotlib.pyplot as plt
+
+from sklearn.metrics import cohen_kappa_score, matthews_corrcoef
+from sklearn.metrics import classification_report
+import numpy as np
+import seaborn
+from collections import defaultdict
+import yaml
+
+
+class TestConfig:
+    def __init__(self, checkpoint_name, device, dataset, train_set, test_set, train_loader, feat_ext, label_cls,
+                 plant_cls, criterion, label_opt, plant_opt, ext_opt, best_loss):
+        self.use_checkpoints = parameters.use_checkpoints
+        self.checkpoint_name = checkpoint_name
+        self.epochs = parameters.epochs
+        self.batch_size = parameters.batch_size
+        self.domain_adapt_lr = parameters.domain_adapt_lr
+        self.device = device
+        self.dataset = dataset
+        self.train_set = train_set
+        self.test_set = test_set
+        self.train_loader = train_loader
+        self.epochs = parameters.epochs
+        self.feat_ext = feat_ext
+        self.label_cls = label_cls
+        self.plant_cls = plant_cls
+        self.criterion = criterion
+        self.label_opt = label_opt
+        self.plant_opt = plant_opt
+        self.ext_opt = ext_opt
+        self.best_loss = best_loss
+        self.loss_delta = parameters.loss_delta
+        self.return_epochs = parameters.return_epochs
+        self.epochs_without_improvement = 0
+
+
+def unique(list1):
+    x = np.array(list1)
+    return list(np.unique(x))
```

Comment:
- [ ] I don't see why this function is needed, can you explain shortly?
- [ ] rewrite to be a bit simpler
```suggestion
def unique(list1):
    return np.unique(list1).tolist()
```